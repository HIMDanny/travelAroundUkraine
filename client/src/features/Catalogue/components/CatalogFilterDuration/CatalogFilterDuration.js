import React, { useEffect } from 'react';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import Grid from '@mui/material/Unstable_Grid2';
import { FilterAccordion, OutlinedToggleButton } from '..';
import { getProducts } from '../../../../store/slices/catalogueSlice/catalogueSlice';


const CatalogFilterDuration = () => {
  const dispatch = useDispatch();
  const products = useSelector((state) => state.catalogue.products, shallowEqual);

  const sorting = () => {
    const duration = [...new Set(products.map((el) => el.duration))];

    const hours = duration
      .filter((el) => el.indexOf('hour') !== -1)
      .sort((a, b) => a.match(/\d+/)[0] - b.match(/\d+/)[0]);

    const days = duration
      .filter((el) => el.indexOf('day') !== -1)
      .sort((a, b) => a.match(/\d+/)[0] - b.match(/\d+/)[0]);

    return [...hours, ...days];
  };

  const sortingToursDuration = sorting();
  
  useEffect(() => {
    dispatch(getProducts());
  }, []);

  return (
    <FilterAccordion title="Duration">
      <Grid container spacing={1}>
        {sortingToursDuration.map((el) => (
          <Grid xs={4} key={el}>
            <OutlinedToggleButton value={el}>{el}</OutlinedToggleButton>
          </Grid>
        ))}
      </Grid>
    </FilterAccordion>
  );
};

export default CatalogFilterDuration;
