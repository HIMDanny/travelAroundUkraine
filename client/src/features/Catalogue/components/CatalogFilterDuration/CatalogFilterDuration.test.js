import { render } from '@testing-library/react';
import CatalogFilterDuration from './CatalogFilterDuration';

jest.mock('@mui/material/ToggleButton', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/material/Unstable_Grid2', () => ({ children }) => <div>{children}</div>);

jest.mock('../FilterAccordion/FilterAccordion', () => ({ children }) => <div>{children}</div>);
jest.mock('../OutlinedToggleButton/OutlinedToggleButton', () => () => <button>test button</button>);

describe('CatalogFilterDuration snapshot test', () => {
  test('should CatalogFilterDuration match snapshot', () => {
    const { asFragment } = render(<CatalogFilterDuration />);
    expect(asFragment()).toMatchSnapshot();
  });
});
