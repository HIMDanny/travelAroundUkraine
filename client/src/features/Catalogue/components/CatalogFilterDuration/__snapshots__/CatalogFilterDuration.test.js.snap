// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogFilterDuration snapshot test should CatalogFilterDuration match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
  </div>
</DocumentFragment>
`;
